### PR TITLE
HDDS-1646. Support real persistence in the k8s example files 

### PR DESCRIPTION
**Pull request by elek**
Ozone release contains example k8s deployment files to make it easier to deploy Ozone to kubernetes. As of now we use emptyDir everywhere, we should support the configuration of host volumes (hostPath or Local Persistent volumes).

The big question here is the default:

* Make the examples easy to start and ephemeral
* Make the examples more safe, by default (but couldn't be started without additional administration).

(Note this conversation is started in the review of HDDS-1508)

Xiaoyu:  Can we support mount hostVolume for datanode daemons?

Marton: Yes, we can.

AFAIK there are two options:
 * using [hostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath)
 * or with [Local PersistentVolumes](https://kubernetes.io/blog/2018/04/13/local-persistent-volumes-beta/)

The first one requires the knowledge of directory names on the host.
The second one is recommended but it requires the creation of PersistentVolumes or install a PersistentVolume provider

I am not sure what is the best approach, my current proposal is:

 * Use empty dir everywhere to make it easier to start simple ozone cluster
 * Provide simple option to turn on any of theses persistence (the kubernetes files are generated and the generation can be parametrized)
 * Document how to customize the kubernetes resources files

Summary: it's question of the defaults:
 
  1. Use a complex, but persistent solution, which may not work out of the box   
  2. Use a simple, but ephemeral solution (as default)

I started to use (2) but I am open to change.

See: https://issues.apache.org/jira/browse/HDDS-1646

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 29 | Docker mode activated. |
||| _ Prechecks _ |
| -1 | dupname | 1 | The patch has 3  duplicated filenames that differ only in case. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-945/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/945 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient yamllint shellcheck shelldocs |
| uname | Linux 307a8431f92e 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / b057479 |
| dupname | https://builds.apache.org/job/hadoop-multibranch/job/PR-945/1/artifact/out/dupnames.txt |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-945/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by elek** [comment]
I created a new kubernetes cluster definition (`getting-started`). After this patch we will have the following resources:

 * minikube (for Minkube usage, ephemeral)
 * getting-started (for real kubernetes cluster, ephemeral)
 * ozone (for normal use-cases, real persistence)
 * ozone-dev (for development and performance test, ephemeral by default)

 * I created rules to add LocalPersistentVolume based PVC
 * I removed the support of daemonset based datanodes as it's not possible to support PVs
 * I added pod antiAffinity rules to ensure that the statefulset based pods are scheduled to different nodes
 * securityContext/fsGroup is used to make sure that we have permission to write the volumes

File: hadoop-ozone/dist/src/main/k8s/definitions/ozone/om-ss.yaml [hunk 1]
```diff
@@ -36,6 +36,8 @@ spec:
         prometheus.io/port: "9874"
         prometheus.io/path: "/prom"
     spec:
+      securityContext:
+        fsGroup: 1000
       initContainers:
       - name: init
         image: elek/ozone
```
Comment (anuengineer) [inline hadoop-ozone/dist/src/main/k8s/definitions/ozone/om-ss.yaml]: I know, not part of this patch. But I hope we have a plan/JIRA to make this point to Ozone dockerhub. Thx

**Comment by anuengineer** [comment]
+1, Thanks for the patch. I will commit this now.
